src/chef-mover/src/mover_server_admins_existing_orgs_permissions_callback.erl

```diff
@@ -0,0 +1,228 @@
+%% -*- erlang-indent-level: 4;indent-tabs-mode: nil; fill-column: 92 -*-
+%% ex: ts=4 sw=4 et
+%% @author Tyler Cloke <tyler@chef.io>
+%% @copyright 2015 Chef Software, Inc.
+%%
+%% This migration iterates through all existing orgs and grants
+%% the server-admins global group READ, UPDATE, and DELETE access on them,
+%% as having access on the container does not update permissions for
+%% existing orgs.
+%%
+%% Should be idempotent. If the admins group doesn't exist for an org,
+%% it creates an empty one to throw server-admins in with a warning.
+
+-module(mover_server_admins_existing_orgs_permissions_callback).
+
+-export([
+         migration_init/0,
+         migration_complete/0,
+         migration_type/0,
+         supervisor/0,
+         migration_start_worker_args/2,
+         migration_action/2,
+         next_object/0,
+         error_halts_migration/0,
+         reconfigure_object/2,
+         needs_account_dets/0
+        ]).
+
+-define(GLOBAL_PLACEHOLDER_ORG_ID, <<"00000000000000000000000000000000">>).
+
+-record(org, {name, id, authz_id}).
+-record(group, {authz_id}).
+-record(mover_chef_group, {
+          id,
+          org_id,	  
+          authz_id,
+          name,
+          last_updated_by,
+          created_at,
+          updated_at
+         }).
+
+migration_init() ->
+    mv_oc_chef_authz_http:create_pool(),
+    mover_transient_migration_queue:initialize_queue(?MODULE, get_orgs()).
+
+
+migration_action(OrgRecord, _AcctInfo) ->
+    OrgName = OrgRecord#org.name,
+    OrgId = OrgRecord#org.id,
+    OrgAuthzId = OrgRecord#org.authz_id,
+    BifrostSuperuserId = mv_oc_chef_authz:superuser_id(),
+    ServerAdminsAuthzId = get_server_admins_authz_id(),
```

Comment:
@markan I'm 👍 on skipping this particular optimization and agree we need to get this merged.  

My thought about saving the sql query was for Hosted where it is less about time and more about the concerns I raised in my June 22 comment. Fewer requests reduces the surface area for transient errors.  In Hosted, per-org migrations that are this complex takes a long time (often through multiple load peaks).  Without the ability to resume, we have to restart the migration from the start if a single thing fails.

I'm still personally nervous about this migration because it looks very similar to the other migrations that got held up in Hosted for a long while.  However, I think the best thing to do is to work to get this merged and address those problems as we hit them.